### PR TITLE
Styling right section of the product details 

### PR DESCRIPTION
**Pull request by momo1113**
* add flexbox on the right sub section make it looks better and responsive
* fetch rating from /reviews and render the average reviews on the page with stars
* customize style' name to colored circles
* add checkmark on each of the style
* implement select style functionality
* implement corresponding selected style with its size options and quantity options
* implement select size and corresponding quantity functionality

### CODE REVIEW

File: ProductDetails/server/routes/productRoutes.js [hunk 1]
```diff
@@ -34,4 +34,24 @@ router.get('/:id/styles', (req, res) => {
...
+    },
+  })
+    .then((response) => {
+      const { results } = response.data;
+      let sum = 0;
+      results.map((result) => {
+        sum += result.rating;
+        return sum;
+      });
+      const aveRating = Math.floor(sum / results.length);
```
Comment (jincheng411) [inline ProductDetails/server/routes/productRoutes.js:50]: According to business requirement, the rating has decimals. Maybe we need to pass the decimals to client side?

**Review by jincheng411 — APPROVED** [review]

File: ProductDetails/src/components/RightSection/TopSection.jsx [hunk 2]
```diff
@@ -1,27 +1,40 @@
...
+  }, []);
+
+  return (
+    <div className="top_section">
+      <Reviews>
+        <StarRatings
+          rating={Number(rating)}
+          starRatedColor="RGB(253, 204, 13)"
+          numberOfStars={5}
+          starDimension={18}
```
Comment (gusfel) [inline ProductDetails/src/components/RightSection/TopSection.jsx:28]: I get this error in the console: Warning: Failed prop type: Invalid prop `starDimension` of type `number` supplied to `StarRatings`, expected `string`.  i think it's about this line.

File: ProductDetails/server/routes/productRoutes.js (hunk 1 above)
Comment (momo1113) [reply ProductDetails/server/routes/productRoutes.js:50]: Good catch Jin. I'll read documentation more carefully.

**Review by gusfel — COMMENTED** [review]
when you select a size and color the drop down with the numbers turns into a list of the same numbers.  not sure if this is something you're aware of and are coming back to.  also, not sure what the numbers are referring to.  is it the number of available products in that size and color?

File: ProductDetails/src/components/RightSection/TopSection.jsx (hunk 2 above)
Comment (momo1113) [reply ProductDetails/src/components/RightSection/TopSection.jsx:28]: I was trying to fix the error. But did't find a right way, just leave on the side. Will try to fix it today. 
Good catch on the list of same numbers. My intention was clicked size should math the quantity on the top of list of quantity list. Need to tweak a little bit that part.

**Review by laenalea — APPROVED** [review]

**Review by gusfel — APPROVED** [review]

**Pull request merged**